If Qt5 is found, its moc will be used even when Qt4 is requested

--- origsrc/cmake-2.8.12.2/Source/QtDialog/CMakeLists.txt	2014-01-16 11:15:08.000000000 -0600
+++ src/cmake-2.8.12.2/Source/QtDialog/CMakeLists.txt	2014-06-30 02:56:22.192515300 -0500
@@ -14,7 +14,9 @@ project(QtDialog)
 if(POLICY CMP0020)
   cmake_policy(SET CMP0020 NEW) # Drop when CMake >= 2.8.11 required
 endif()
+if(NOT QT_QMAKE_EXECUTABLE MATCHES ".*qt4.*")
 find_package(Qt5Widgets QUIET)
+endif()
 if (Qt5Widgets_FOUND)
   include_directories(${Qt5Widgets_INCLUDE_DIRS})
   add_definitions(${Qt5Widgets_DEFINITONS})
